File: 백준/Silver/11047. 동전 0/동전 0.py
# N개 종류의 동전, 종류 별로 많이 가지고 있음
# 적절히 동전을 섞어서 합쳐서 K원이 나와야 됨.
# 필요한 동전의 최솟값을 구하시오

# N >> 동전 종류 개수
# K >> 동전 합쳐서 나와야 되는 결과값
N, K = map(int, input().split());

# num_list 동전 종류 리스트
num_list = [];

for i in range(N):
    num_list.append(int(input()));

num_list.sort(reverse=True);

answer = 0;

for i in num_list:
    answer += int(K /i);
    K = K % i;    

print(answer);

File: 백준/Silver/11047. 동전 0/동전 0.js
const fs = require('fs');
const filePath = process.platform === 'linux' ? 'dev/stdin' : '../input.txt';
const input = fs.readFileSync(filePath).toString().trim().split('\n');

//동전을 N개 종류를 가지고 있음. >> 각 동전마다 많이 가지고 있음
//동전을 적절히 사용해서 그 가치의 합을 K로 만들려고 함.

let [N, K] = input[0].split(' ').map((i) => parseInt(i));

let coin = [];

for (let i = 1; i <= N; i++) {
  coin.push(parseInt(input[i]));
}

coin = coin.sort((a, b) => b - a);

let count = 0;

for (let i = 0; i < N; i++) {
  count += parseInt(K / coin[i]);
  K %= coin[i];
}

console.log(count);
